src/one_dragon/envs/git_service.py

```
@@ -27,6 +29,35 @@
+    Returns:
+        操作返回值
+
+    Raises:
+        TimeoutError: 操作超时
+    """
+    with concurrent.futures.ThreadPoolExecutor(max_workers=1) as executor:
+        future = executor.submit(fn)
+        try:
+            return future.result(timeout=timeout)
+        except concurrent.futures.TimeoutError:
+            raise TimeoutError(f'{operation_name} 超时（{timeout}秒）')
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)